neo4j/_async/io/_pool.py

```diff
@@ -370,6 +403,22 @@ def __repr__(self):
         """
         return "<{} addresses={!r}>".format(self.__class__.__name__, self.get_default_database_initial_router_addresses())
 
+    @asynccontextmanager
+    async def _refresh_lock_deadline(self, deadline):
+        timeout = deadline.to_timeout()
+        if timeout == float("inf"):
+            timeout = -1
+        if not await self.refresh_lock.acquire(timeout=timeout):
+            raise ClientError(
+                "Failed to wait for the pool to updated the routing "
+                "table within {!r}s".format(deadline.original_timeout)
```

Comment:
Pool failed to update routing table within timeout.
```suggestion
                "Pool failed to update routing table within "
                "{!}s timeout.".format(deadline.original_timeout)
```

Reply:
Addressed :writing_hand: